src/main/kotlin/org/pkl/lsp/analyzers/ModuleUriAnalyzer.kt

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pkl.lsp.analyzers
+
+import java.net.URI
+import java.net.URISyntaxException
+import org.pkl.lsp.*
+import org.pkl.lsp.actions.PklDownloadPackageAction
+import org.pkl.lsp.ast.*
+import org.pkl.lsp.packages.PackageDependency
+import org.pkl.lsp.packages.dto.Checksums
+import org.pkl.lsp.packages.dto.PackageUri
+import org.pkl.lsp.packages.dto.Version
+
+class ModuleUriAnalyzer(project: Project) : Analyzer(project) {
```

Comment:
I changed `ImportAnalyzer` to this because amends/extends headers aren't imports.